fineract-provider/src/main/java/org/apache/fineract/portfolio/note/domain/Note.java

```diff
@@ -63,107 +72,17 @@ public class Note extends AbstractAuditableWithUTCDateTimeCustom<Long> {
     private Integer noteTypeId;
```

Comment:
Could we use the enum itself in the entity and map it as @Enumerated(ORDINAL)? Wouldn't it achieve the same results, but allow to use the enum from the Java code side?

Reply:
Yes, I think it would be better, and also I would recommend using `EnumType.STRING` instead to avoid any conflict if we change the order or something.

But I think it's out of the scope of this PR for now.

Reply:
Unfortunately we cant change this right now, because this Integer is not the ordinal of the enum, but a custom number associated with each enum type. This topic would worth a discussion and then a refactor of the more hundred enum-like properties not using real enum types, but this is really not the scope of this PR.